### PR TITLE
trigger rotation correctly if files laid down by the startup container change

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
NONE
```

The short term goal here is to include the expected output of the startup container into the master hash.

### CODE REVIEW

File: pkg/cluster/hash.go [hunk 1]
```diff
@@ -10,51 +10,81 @@ import (
...
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.NewEncoder(hash).Encode(vmss)
+	if err != nil {
+		return nil, err
+	}
+
+	err = arm.WriteStartupFiles(h.log, cs, writers.NewTarWriter(hash), config.GetHostname(app, "", instanceID), "")
```
Comment (mjudeikis) [inline pkg/cluster/hash.go]: I imagine this is using same writer to append data to it?
Was this main motivation to move to writer based function?

**Comment by mjudeikis** [comment]
Despite the fact that most of these are cosmetic changes (move stuff around, rename, etcd). It is really hard to notice some possible errors in the new hash'er implementation. Overall looks good. can lgtm, if anybody else dont want to take a look.

File: pkg/cluster/hash.go (hunk 1 above)
Comment (jim-minter) [reply pkg/cluster/hash.go]: yes, all the writer work is leading up to this line :)

**Comment by jim-minter** [comment]
lgtm it, there's more where this came from...

**Pull request merged**